scripts/perftest/ohe_checks/README.md

```diff
@@ -0,0 +1,151 @@
+
+**Candidate Processing:** Within the loop, after adding a column to `currentCandidates`:
+- Calls `isHotEncoded` to check the status of the candidates.
+- If `isHotEncoded` returns `"NOT_OHE"`, moves the candidates to regular groups and clears the candidates list.
+- If `isHotEncoded` returns `"VALID_OHE"`, moves the candidates to `oheGroups` and clears the candidates list.
+- If `isHotEncoded` returns `"POTENTIAL_OHE"`, continues accumulating candidates.
+
+### 6. Final Candidate Check
+
+**Post-loop Check:** After the loop, checks any remaining `currentCandidates`:
+- If they form a valid OHE group, adds them to `oheGroups`.
+- Otherwise, adds them to regular groups.
+
+### 7. Overwrite and CoCode Groups
+
+**Overwrite Groups:** Updates `colInfos.compressionInfo` with the processed `groups`.
+**OHE Group Integration:** Combines indexes for validated OHE groups and adds them to the final `groups`.
+
+## One Hot Encoded Columns Compression in `ColGroupFactory`
+
+### Description
+
+The `compressOHE` function is designed to compress columns that are one-hot encoded (OHE). It validates and processes the input data to ensure it meets the criteria for one-hot encoding, and if so, it compresses the data accordingly. If the data does not meet the OHE criteria, it falls back to a direct compression method (`directCompressDDC`).
+
+### Implementation Details
+
+1. **Validation of `numVals`**:
+   - Ensures the number of distinct values (`numVals`) in the column group is greater than 0.
+   - Throws a `DMLCompressionException` if `numVals` is less than or equal to 0.
+
+2. **Handling Transposed Matrix**:
+   - If the matrix is transposed (`cs.transposed` is `true`):
+     - Creates a `MapToFactory` data structure with an additional unique value.
+     - Iterates through the sparse block of the matrix, checking for non-one values or multiple ones in the same row.
+     - If a column index in the sparse block is empty, or if non-one values or multiple ones are found, it falls back to `directCompressDDC`.
+
+3. **Handling Non-Transposed Matrix**:
+   - If the matrix is not transposed (`cs.transposed` is `false`):
+     - Creates a `MapToFactory` data structure.
+     - Iterates through each row of the matrix:
+       - Checks for the presence of exactly one '1' in the columns specified by `colIndexes`.
+       - If multiple ones are found in the same row, or if no '1' is found in a sample row, it falls back to `directCompressDDC`.
+
+4. **Return Value**:
+   - If the data meets the OHE criteria, returns a `ColGroupDDC` created with the column indexes, an `IdentityDictionary`, and the data.
+   - If the data does not meet the OHE criteria, returns the result of `directCompressDDC`.
+
+## Add method in `ColGroupSizes`
+Added method ``estimateInMemorySizeOHE(int nrColumns, boolean contiguousColumns, int nrRows)``
+
+## Add method in `AComEst`
+Added a getter method `getNnzCols`
+
+## Edit `distinctCountScale` method in `ComEstSample`
```

Comment:
all these edit details should not be here, we should just make the fixes to the system in general instead.